src/utils/transition-manager.js

```diff
@@ -142,6 +144,16 @@ export default class TransitionManager {
     return false;
   }
 
+  _normalizeNextProps(nextProps: ViewportProps, startProps: ViewportProps): ViewportProps {
+    const endProps = Object.assign({}, nextProps);
+    const {transitionInterpolator} = nextProps;
+    if (transitionInterpolator) {
+      const transitionDuration = transitionInterpolator.getDuration(startProps, nextProps);
```

Comment:
This is called on every frame - can we move the `getDuration` call to `_triggerTransition`?

Reply:
Updated to make this call during `_triggerTransition`,  also updated `_isTransitionEnabled` to check for `auto` as that is called before actual duration is calculated.

Reply:
You should not need to validate `transitionInterpolator` again since transition cannot be triggered without it. I would merge this one-line code with `_triggerTransition`.

Reply:
Updated.